source/Halibut.TestProxy/HttpProxyService.cs

```diff
@@ -39,45 +41,78 @@ public HttpProxyService(HttpProxyOptions options, ILoggerFactory loggerFactory)
             this.logger = loggerFactory.CreateLogger<HttpProxyService>();
         }
 
-        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        public async Task StartAsync()
         {
+            var _ = Task.Run(async () =>
```

Comment:
On a second look this is going to hang if `StartInternalAsync` fails before `started == true`. Will revisit and ensure it bails in that condition